### ISSUE TITLE
"Extend > Reducing CSS Size". Not always the case.

### ISSUE THREAD

**Issue by maniqui** [less/less-docs#177]
Using the `:extend(.selector all)` pseudo-selector _not always_ provides a shorter output (ie. smaller filesize), compared to the output generated when using the same selector (the one being `:extend()ed`) as a mixin. 

Of course, this could be expected as using `:extend(.selector all)` is _similar_ but _not exactly the same_ to using a mixing.
Why? 
Because when using  `:extend(.selector all)`, it even extends the selector when it's used in more complex selectors (for example, as a child, like in `.parent .selector`). 
Using a mixin doesn't cover those cases.

A quick example:

```
// Input, using  :extend(.a all):
.a {
  prop: value;

  &:hover,
  &:focus {
    prop: value;
  } 
}

.parent .a { // Here is a "complex" selector which is also extended.
  prop: value;

  &:hover,
  &:focus {
    prop: value;
  }
}

.some-selector {
  &:extend(.a all); // We extend .a, everywhere it's used as a selector.
}
```

``` css
// Output:
.a,
.some-selector {
  prop: value;
}
.a:hover,
.a:focus,
.some-selector:hover,
.some-selector:focus {
  prop: value;
}
.parent .a,
.parent .some-selector {
  prop: value;
}
.parent .a:hover,
.parent .a:focus,
.parent .some-selector:hover,
.parent .some-selector:focus {
  prop: value;
}
```

**Output length: 289 characters**

Now, a similar use case, but using `.a` as a mixin.

```
// Input
.a {
  prop: value;

  &:hover,
  &:focus {
    prop: value;
  }

}

.parent .a { // This one will not be "mixed in".
  prop: value;

  &:hover,
  &:focus {
    prop: value;
  }
}

.some-selector {
  .a;
}
```

``` css
// Output
.a {
  prop: value;
}
.a:hover,
.a:focus {
  prop: value;
}
.parent .a {
  prop: value;
}
.parent .a:hover,
.parent .a:focus {
  prop: value;
}
.some-selector {
  prop: value;
}
.some-selector:hover,
.some-selector:focus {
  prop: value;
}
```

**Output length: 242 characters**

As you can see, the version using `:extend` is some bytes larger (and, let me state it again, it include some extra rules not present in the mixin output).

In case you are looking for some _real case examples_, I suggest you to try extending and mixing in Bootstrap, using the classes `.btn` and `.panel`, for example. 
Extending the `.btn` class will get you a far more larger output than using it as a mixin. Of course, when extending, there are more selectors "covered" than when mixin in.
While doing the same for the `.panel` class leads to a shorter output when using `:extend(.panel all)` than when using it as a .mixin.

TL;DR: using `:extend()` usually leads to shorter CSS output, but that's not always the case.
